src/api/search/test/query.test.js

```diff
@@ -1,9 +1,173 @@
+    mock.add(
+      {
+        method: ['POST', 'GET'],
+        path: '/posts/post/_search',
+      },
+      () => {
+        return {};
+      }
+    );
+    const res = await request(app).get('/').query({ text: '', filter: 'post' });
+    expect(res.statusCode).toBe(400);
+    expect(res.body).toStrictEqual([
+      { location: 'query', msg: 'text should not be empty', param: 'text', value: '' },
+    ]);
+  });
+
+  it('return 200 with empty results if text is given ""', async () => {
+    mock.add(
+      {
+        method: ['POST', 'GET'],
+        path: '/posts/post/_search',
+      },
+      () => {
+        return {
+          results: 0,
+          hits: {
+            total: { value: 0 },
+            hits: [],
+          },
+        };
+      }
+    );
+    const res = await request(app).get('/').query({ text: '""', filter: 'post' });
+    expect(res.statusCode).toBe(200);
+    expect(res.body).toStrictEqual({ results: 0, values: [] });
+  });
+
+  it('return 200 if filter by post and given text', async () => {
+    mock.add(
+      {
+        method: ['POST', 'GET'],
+        path: '/posts/post/_search',
+      },
+      () => {
+        return {
+          results: 2,
+          hits: {
+            total: { value: 2 },
+            hits: [
+              {
+                _id: '1234',
+                url: `${POSTS_URL}/`,
+              },
+              {
+                _id: '5678',
+                url: `${POSTS_URL}/`,
+              },
+            ],
+          },
+        };
+      }
+    );
+    const res = await request(app).get('/').query({ text: 'Telescope', filter: 'post' });
+    expect(res.statusCode).toBe(200);
+    expect(res.body.results).toBe(2);
+    expect(res.body.values).toStrictEqual([
+      { id: '1234', url: `${POSTS_URL}/1234` },
+      { id: '5678', url: `${POSTS_URL}/5678` },
+    ]);
+  });
+
+  it('return 200 if filter by author and given text', async () => {
+    mock.add(
+      {
+        method: ['POST', 'GET'],
+        path: '/posts/post/_search',
+      },
+      () => {
+        return {
+          results: 2,
+          hits: {
+            total: { value: 2 },
+            hits: [
+              {
+                _id: '9966',
+                url: `${POSTS_URL}/`,
+              },
+              {
+                _id: '1122',
+                url: `${POSTS_URL}/`,
+              },
+            ],
+          },
+        };
+      }
+    );
+    const res = await request(app).get('/').query({ text: 'Bob', filter: 'author' });
+    expect(res.statusCode).toBe(200);
+    expect(res.body.results).toBe(2);
+    expect(res.body.values).toStrictEqual([
+      { id: '9966', url: `${POSTS_URL}/9966` },
+      { id: '1122', url: `${POSTS_URL}/1122` },
+    ]);
+  });
+
+  it('return 503 if server has error', async () => {
+    mock.add(
+      {
+        method: ['POST', 'GET'],
+        path: '/posts/post/_search',
+      },
+      () => {
+        return { statusCode: 503 };
```

Comment:
I mentioned in my tests as well, it's not the `statusCode` that is being returned by ElasticSearch that is returning the 503 error. 
It's failing because Search is trying to do [these two lines](https://github.com/Seneca-CDOT/telescope/blob/f5785c6c33e4747260aab267d9697aa105fb32bc/src/api/search/src/search.js#L75-L78) here in the return.
It's trying to read `hits.total` and `hits.hits`,  but the `hits` object is undefined here because its not being returned by ElasticSearch. 
For example, changing it to `return { statusCode: 404 }` will also pass this test. 
I think we might need to rephrase the description for this test, so its not as misleading

Reply:
Should I try to return something that fails to read the `hits` object?